### PR TITLE
`yvals_core.h`: Consistent diagnostics and warnings

### LINKED ISSUE #237: <yvals_core.h>: "acknowledge this warning" vs. "suppress this deprecation"
In `<yvals_core.h>`, we define deprecation messages of the form "You can define MEOW to suppress this warning" and messages of the form "You can define MEOW to acknowledge this warning." The former form should be used consistently for all deprecation warnings, and the latter for all non-deprecation warnings.

### PR DESCRIPTION
**Pull request by sam20908**
Fixes #237
Fixed broken PR history of #2969

### CODE REVIEW

**Review by strega-nil-ms — COMMENTED** [review]
You should also change anything that says "suppress this deprecation" to "suppress this warning".

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]

File: stl/inc/yvals_core.h
Comment (sam20908) [reply stl/inc/yvals_core.h:742]: Agreed.

File: stl/inc/yvals_core.h
Comment (sam20908) [inline stl/inc/yvals_core.h:752]: Ditto above point with acknowledge being inaccurate.

File: stl/inc/yvals_core.h
Comment (sam20908) [reply stl/inc/yvals_core.h:727]: I'm going to keep `suppress` here for the consistency of warnings. If others feel that's not the case then I can change it.

File: stl/inc/yvals_core.h
Comment (sam20908) [inline stl/inc/yvals_core.h:1549]: Ditto above point with acknowledge being inaccurate.

**Review by strega-nil-ms — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Marking as `decision needed` for the reasons that I explained here: https://github.com/microsoft/STL/issues/237#issuecomment-548190058 (I suspect I made that comment before we created the `decision needed` label.)

Also, there are occurrences of "acknowledge" outside `<yvals_core.h>`. If we decide that this should be changed, the occurrences in `<hash_map>`, `<hash_set>`, `<type_traits>`, `<experimental/coroutine>`, and `<experimental/filesystem>` should be checked.

**Comment by StephanTLavavej** [comment]
We talked about this at the weekly maintainer meeting - the consensus was that we should consistently say `suppress`, and that we should refer to things as `warning`s or `error`s but not `diagnostic`s (as that's Standard jargon, not in wide use among programmers).

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks! Everything in `<yvals_core.h>` looks good. Searching for all occurrences of "acknowledge" and "diagnostic", I found a few more files that needed to be updated (I previously mentioned some of them):

* Say "suppress this error" for "hard deprecations".
* Say "suppress this error" for the Clang experimental/coroutine message.
  + Drop "and acknowledge that this is unsupported"; this repeats "currently do not support Clang".
* Reword the `aligned_storage` message.
  + Unify "acknowledge that you understand this message and" and "silence this message and" into "To suppress this error,".
  + Say "confirm" and drop "actually".
* Change "diagnostic" to "error" in the `join_view` message.

FYI @strega-nil-ms as I pushed these changes after you approved (I believe they all align with what you wanted).

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for making these messages simpler and more consistent! :speech_balloon: :tada: :smile_cat:

**Pull request merged**
